src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java

```diff
@@ -247,38 +246,83 @@ private int chunkMapType(
+            } else {
+                // chunk the object when target key is of leaf type (null, string and list of string)
+                Object chunkObject = sourceAndMetadataMap.get(originalKey);
+                stringTobeChunkedCount += getStringTobeChunkedCountFromLeafType(chunkObject);
+            }
+        }
+        return stringTobeChunkedCount;
+    }
+
+    @SuppressWarnings("unchecked")
+    private int getStringTobeChunkedCountFromLeafType(final Object value) {
+        // leaf type means null, String or List<String>
+        // the result should be an empty list when the input is null
+        if (value instanceof String) {
+            return StringUtils.isEmpty((String) value) ? 0 : 1;
+        } else if (isListOfString(value)) {
+            return (int) ((List<String>) value).stream().filter(s -> !StringUtils.isEmpty(s)).count();
+        }
+        return 0;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void chunkMapType(
+        Map<String, Object> sourceAndMetadataMap,
+        final Map<String, Object> fieldMap,
+        final Map<String, Object> runtimeParameters
+    ) {
+        for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
+            String originalKey = fieldMapEntry.getKey();
+            Object targetKey = fieldMapEntry.getValue();
+            if (targetKey instanceof Map) {
+                // call this method recursively when target key is a map
+                Object sourceObject = sourceAndMetadataMap.get(originalKey);
+                if (sourceObject instanceof List) {
+                    List<Object> sourceObjectList = (List<Object>) sourceObject;
+                    for (Object source : sourceObjectList) {
+                        if (source instanceof Map) {
+                            chunkMapType((Map<String, Object>) source, (Map<String, Object>) targetKey, runtimeParameters);
+                        }
+                    }
+                } else if (sourceObject instanceof Map) {
+                    chunkMapType((Map<String, Object>) sourceObject, (Map<String, Object>) targetKey, runtimeParameters);
+                }
             } else {
                 // chunk the object when target key is of leaf type (null, string and list of string)
                 Object chunkObject = sourceAndMetadataMap.get(originalKey);
                 List<String> chunkedResult = chunkLeafType(chunkObject, runtimeParameters);
                 sourceAndMetadataMap.put(String.valueOf(targetKey), chunkedResult);
             }
         }
-        return updatedChunkCount;
     }
 
     /**
      * Chunk the content, update the runtime max_chunk_limit and return the result
      */
     private List<String> chunkString(final String content, final Map<String, Object> runTimeParameters) {
-        // update runtime max_chunk_limit if not disabled
+        // return an empty list for empty string
+        if (StringUtils.isEmpty(content)) {
+            return List.of();
```

Comment:
What's this empty list used for?

Reply:
If we do not return empty list directly, stringTobeChunkedCount will be reduced by 1. This is not expected because only non-empty will take up max_chunk_limit. This parameter is thus ignored for empty string.